### ISSUE TITLE
[Feature request] Ask user to verify the amount is correct when sending a "large" amount

### ISSUE THREAD

**Issue by dmp1ce** [mycelium-com/wallet#217]
I recently accidently copied my amount into the BTC instead of USD. Usually, when I do that the amount is too large and it won't send. Recently that didn't happen and I sent about 320 times too much!

It would be great if there was an additional prompt for sending above a certain threshold. Perhaps the threshold could be set in the settings. Something like, "You are sending $X over your threshold. Are you sure you want to send $Y?"

**Comment by Giszmo** [comment]
I would also like to fix the problem but I'm not sure it's easy to fix without being annoying. I would be happy already to see the amount during pin entry and users that have no pin set. Assuming that users with serious amounts in the wallet have a pin set, that would help a lot I guess.

**Comment by FrancisPouliot** [comment]
Yes, same thing happened to us at our exchange. An employee accidentally scanned a wrong address and sent a large amount to the wrong address.

This is a feature of mytrezor.com also by the way, would be great to have implemented in Mycelium.

Maybe: Settings - Pin settings - Second PIN for large amounts (select amount) ?

**Comment by Transisto** [comment]
I'd add the option to enable asking for a challenge when sending "large" amount, like typing the fist 4 character of the receiver bitcoin address.  The main error that happen is sending to the wrong address.  Reflexes build up and bam!, lose insane amount of money.

**Comment by dmp1ce** [comment]
@Giszmo I have a PIN set but I just wasn't paying close enough attention. I meant to send $5 but instead sent $1600, which was almost everything in my wallet! I never send more than $100 usually, so $100 would have been a nice threshold for me.

**Comment by jonathancross** [comment]
@dmp1ce: It seems that you typed "5" expecting "_5 USD_", but the app was defaulting to BTC currency, correct?  Better handling of default local currency per #269 might reduce the occurrences of such mistakes.

**Comment by dmp1ce** [comment]
@jonathancross That might have helped. I probably would have had USD as the default.

**Comment by Giszmo** [comment]
It doesn't solve the problem. Next somebody tries to send bits and finds out he sent $1000 instead of mere 1000bits.

I guess it's reasonable to have some limit the user can bump up. Like "Seams you are trying to send money {for the first time|above $10|above $100|above $1k|…}. Once sent, it's gone. Are you sure this is what you want? Transaction details: …"

**Comment by Giszmo** [comment]
The "high amount" being sent can also be due to the mining fee. Added tag.

**Comment by dmp1ce** [comment]
Mining fees probably wouldn't bump the expected amount by 2500 to 3000 times though, like it would if you made my mistake now!

Since the value is so high now, I don't really have the same issue anymore. When I did have the issue, I just created an account for small transactions where I had less than 1 or 2 bitcoins. I guess this could still happen today if you wanted to send someone $1 and you accidentally sent them 1 bitcoin!
